pkg/webhook/webhook.go

```diff
@@ -504,7 +526,48 @@ func MutateHandler(w http.ResponseWriter, req *http.Request) {
 				})
 			}
 
-			patch = createVolPatch(patch)
+			// Determine if hugepages are being requested for a given container,
+			// and if so, expose the value to the container via Downward API.
+			hugepageResourceList := make(map[string]hugepageResourceData)
+			glog.Infof("injectHugepageDownApi=%v", injectHugepageDownApi)
+			if injectHugepageDownApi {
+				for _, container := range pod.Spec.Containers {
+					if len(container.Resources.Requests) != 0 {
+						if quantity, exists := container.Resources.Requests["hugepages-1Gi"]; exists && quantity.IsZero() == false {
+							hugepageResourceList["requests.hugepages-1Gi"] = hugepageResourceData{
+								ContainerName: container.Name,
+								Path:          types.Hugepage1GiRequestPath,
+								DivisorVal:    1*1024*1024*1024,
```

Comment:
How is the quantity of the request exposed to pod? Number of 1Gi huge pages for example? I don't see it here.

Reply:
It will show up as it is reserved. For 1G hugepages for example. where 2G has been requested:
```
sh-4.2# cat /etc/podnetinfo/req_hugepages_1Gi 
2
sh-4.2# cat /etc/podnetinfo/lim_hugepages_1Gi 
2
```
Need to make sure 2M is correct.

Reply:
2M works as coded, for 2G at 2M hugepage size,
```
sh-4.2# cat /etc/podnetinfo/hugepages_2Mi_request 
2000
sh-4.2# cat /etc/podnetinfo/hugepages_2Mi_limit
2000
```
However, now that you have me thinking about it, the container doesn't care if it is 1G or 2M hugepage size, DPDK for example just uses `-m 1024`. So I am going to slightly rework so that the value is constant and can just be applied. Also, see pending reply to your other comment about standard path.